autowheelbarrow.lua

```diff
@@ -0,0 +1,120 @@
+                   flags[21] == true or flags[22] == true or flags[23] == true or flags[24] == true or
+                   flags[25] == true or flags[26] == true or flags[27] == true or flags[28] == true or
+                   flags[29] == true or flags[30] == true or flags[31] == true then
+                    skip_wheelbarrows = true
+                    --dfhack.println("individual stockpiles Skipping stockpile due to one or more flags being true at " .. tostring(building))
+                end
+            end
+
+            -- Check if all flags are set to false (for the none stockpile)
+            if stockpile_settings and stockpile_settings.flags then
+                local flags = stockpile_settings.flags
+                if flags.animals == false and flags.food == false and flags.furniture == false and
+                   flags.corpses == false and flags.refuse == false and flags.stone == false and flags.ammo == false and
+                   flags.coins == false and flags.bars_blocks == false and flags.gems == false and flags.finished_goods == false and
+                   flags.leather == false and flags.cloth == false and flags.wood == false and flags.weapons == false and
+                   flags.armor == false and flags.sheet == false and
+                   flags[17] == false and flags[18] == false and flags[19] == false and flags[20] == false and
+                   flags[21] == false and flags[22] == false and flags[23] == false and flags[24] == false and
+                   flags[25] == false and flags[26] == false and flags[27] == false and flags[28] == false and
+                   flags[29] == false and flags[30] == false and flags[31] == false then
+                    skip_wheelbarrows = true
+                    --dfhack.println("Skipping NONE stockpile" .. tostring(building))
+                end
+            end
+
+            if skip_wheelbarrows then
+                building.max_wheelbarrows = 0
+            else
+                local count = unassign_wheelbarrows()
+                building.max_wheelbarrows = count
+                --dfhack.println("Set wheelbarrow limit for stockpile at " .. tostring(building) .. " to " .. count)
+            end
+        end
+    end
+end
+
+local function auto_wheelbarrows()
+    for _, unit in ipairs(df.global.world.units.active) do
+        if dfhack.units.isCitizen(unit) and unit.job.current_job and unit.job.current_job.job_type == 38 then
+            unassign_wheelbarrows()
+            set_wheelbarrows_for_all_stockpiles()
+            return
+        end
+    end
+end
+
+-- Event loop function to call periodically
+local function event_loop()
+    if enabled then
+        dfhack.println("Running autowheelbarrow.lua ")
+        auto_wheelbarrows()
+        -- Check again in 7 days
+        repeatutil.scheduleUnlessAlreadyScheduled(GLOBAL_KEY, 7, "days", event_loop)
+    end
+end
```

Comment:
The reason for the timing of seven days is children decide to pick up a wheelbarrow and it can take them quite a bit of time to haul the wheelbarrow meaning if the child is hauling for a while this will cause the program to run again and unassign all of the wheelbarrows in the fort resulting in a loop causing less work to be done.

Reply:
The 7 days may be fine, even though it may be preferable to make this configurable. My question was more: why should the reassignment only occur if there is an active `StoreItemInStockpile` job.